### PR TITLE
[MAINTENANCE] GX-Release Process update to use GitHub Actions

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Shinnnyshinshin) [inline .github/workflows/ci.yml]: require all of the other stages to pass before we `build-n-publish`

File: .github/workflows/ci.yml
Comment (Shinnnyshinshin) [inline .github/workflows/ci.yml:401]: if we have pushed a tag

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml:401]: https://github.com/orgs/community/discussions/25692

File: setup.cfg
Comment (Kilo59) [inline setup.cfg]: Do you have a link or quick summary explaining this change?

Comment (Shinnnyshinshin) [reply setup.cfg]: oh oops that was a left-over from debugging. I should clean it up

File: ci/azure-pipelines.yml
Comment (Shinnnyshinshin) [inline ci/azure-pipelines.yml:549]: commented out the `deploy` stage for now

File: .github/workflows/ci.yml
Comment (billdirks) [inline .github/workflows/ci.yml]: nit: can you restore this blank line. I like having a blank line between jobs and having jobs have no blank lines in them. It makes it easier for me to parse quickly.

File: .github/workflows/ci.yml
Comment (billdirks) [reply .github/workflows/ci.yml]: nit: Can you format this like I did for `notify_on_failure`, ie one dependency per line.

File: .github/workflows/ci.yml
Comment (billdirks) [reply .github/workflows/ci.yml:401]: I don't think you need this because you filter the push requests to only tags. But this is ok also.

**Review by billdirks — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (cdkini) [inline .github/workflows/ci.yml]: Does this know to release a particular version? If I tag `0.17.x`, will that be the distribution created?

Does this prevent against duplicates? The docs for this action has a section on that - mind just reading up to make sure?

File: .github/workflows/ci.yml
Comment (cdkini) [inline .github/workflows/ci.yml:18]: Nit - It would be good to have a top-level docstring explaining the different cases that this CI is run. Not necessary for this PR but would be a nice way to wrap up the effort imo

File: .github/workflows/ci.yml
Comment (cdkini) [inline .github/workflows/ci.yml]: Let's flip these values to prod before merging and try the actual release with this script.

If you're okay with that, let's comment out the release step of the Azure pipeline as part of this PR.

File: .github/workflows/ci.yml
Comment (cdkini) [reply .github/workflows/ci.yml:18]: Although I guess we have some good comments above - will leave it up to you and the rest of the team

File: .github/workflows/ci.yml
Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml]: done

**Pull request merged**
